File: backend/ms-properties/src/main/java/pi/ms_properties/domain/Inquiry.java
package pi.ms_properties.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name = "Inquiry")
public class Inquiry {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = true)
    private String userId;

    @Column(name = "phone", nullable = false)
    private String phone;

    @Column(name = "email", nullable = false)
    private String email;

    @Column(name = "first_name", nullable = false)
    private String firstName;

    @Column(name = "last_name", nullable = false)
    private String lastName;

    @Column(name = "date", nullable = false)
    private LocalDateTime date;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "description", nullable = false)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private InquiryStatus status;

    @Column(name = "date_close", nullable = true)
    private LocalDateTime dateClose;

    @ManyToMany(mappedBy = "inquiries")
    @JsonIgnore
    private List<Property> properties = new ArrayList<>();
}

File: frontend/src/app/property/types/inquiry.ts
/** Enum equivalente al InquiryStatus del back-end */
export type InquiryStatus = "ABIERTA" | "CERRADA";

export interface InquiryCreateAuth {
  userId?: string;
  title: string;
  description: string;
  propertyIds?: number[];
}

/** Lo que envía un usuario NO autenticado */
export interface InquiryCreateAnon extends InquiryCreateAuth {
  email: string;
  phone: string;
  firstName: string;
  lastName: string;
}

export interface Inquiry {
  id: number;
  userId?: string;
  phone: string;
  email: string;
  firstName: string;
  lastName: string;
  date: string;
  title: string;
  description: string;
  status: InquiryStatus;
  dateClose?: string;
  propertyTitles: string[];
}
